### PR TITLE
Adding the Identity Management Nevermined spec

### PR DESCRIPTION
**Pull request by aaitor**
## Description

Add a description of your changes here.

## Is this PR related with an open issue?

Related to Issue #

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 
## Checklist:

- [ ] Follows the code style of this project.
- [ ] Tests Cover Changes
- [ ] Documentation

### CODE REVIEW

File: docs/architecture/specs/id_management/README.md [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+* Users managing permissions on these decentralized assets should be able to add
+  and revoke permissions
+* The schemas and data existing in the individual deployments of the Identity
+  Services shouldn't be replicated in the decentralized ecosystem. The source of
+   truth are the Identity Services instances
+* Information kept about the users and groups in the DLT network must be minimal
+  and never store PI. Ideally only ids and hashes
+* The decentralized ecosystem register assets and the conditions for who can
+  interact and what is possible to do with these assets is kept on-chain
+* Public information about users like users public keys should be available
```
Comment (r-marques) [inline docs/architecture/specs/id_management/README.md:168]:
```suggestion
* Public information about users like users public keys should be available.
```

File: docs/architecture/specs/id_management/README.md [hunk 2]
```diff
@@ -0,0 +1,462 @@
...
+* "cty" - Header Parameter defined by JSON Web Signature and JSON Web Encryption
+ is used by this specification to convey structural information about the JWT.
+
+In our case the header is gonna look like this:
+```json
+{
+  "typ": "JWT"
+  "alg": "ES256",
+  "cty": "arbitrary"
+}
```
Comment (r-marques) [inline docs/architecture/specs/id_management/README.md:285]:
```suggestion
{
  "typ": "JWT",
  "alg": "ES256",
  "cty": "arbitrary"
}
```

File: docs/architecture/specs/id_management/README.md [hunk 3]
```diff
@@ -0,0 +1,462 @@
...
+* Private claims: These are the custom claims created to share information
+  between parties that agree on using them.
+
+In our case the payload will looks like this:
+```json
+{
+  "iss": "0x123456"
+  "sub": "did:nv:abcde",
+  "aud": "0xffffff"
+}
```
Comment (r-marques) [inline docs/architecture/specs/id_management/README.md:307]:
```suggestion
{
  "iss": "0x123456",
  "sub": "did:nv:abcde",
  "aud": "0xffffff"
}
```

File: docs/architecture/specs/id_management/README.md [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+
+* Reserved claims: These are a set of predefined claims, which are not mandatory
+  but recommended, thought to provide a set of useful, interoperable claims.
+* Public claims: These can be defined at will by those using JWTs. But to avoid
+  collisions they should be defined in the IANA JSON Web Token Registry or be
+  defined as a URI that contains a collision resistant namespace.
+* Private claims: These are the custom claims created to share information
+  between parties that agree on using them.
+
+In our case the payload will looks like this:
```
Comment (r-marques) [inline docs/architecture/specs/id_management/README.md:301]: maybe its worth defining what the payload fields are the same way it was done for the header

File: docs/architecture/specs/id_management/README.md [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+
+
+## Annex
+
+### Serializing DID Documents
+
+Having a standard JSON document or subtract, the common operations used to
+serialize fragments of DID Documents is as follows:
+
+* The object is sorted alphabetically independently of the existing nested levels
```
Comment (r-marques) [inline docs/architecture/specs/id_management/README.md:454]:
```suggestion
* The object is sorted alphabetically by key, independently of the existing nested levels
```

**Review by r-marques — APPROVED** [review]
some comments but lgtm

**Comment by aaitor** [comment]
changes requested applied

**Pull request merged**